Source/BestProjectEver/Private/HUD/BPE_HUD.cpp

```diff
@@ -83,10 +93,86 @@ void ABPE_HUD::UpdateHealth(const FHealthData& HealthData)
 }
 
 //----------------------------------------------------------------------------------------------------------------------
-void ABPE_HUD::UpdateMatchTimer(const float TimeLeft)
+void ABPE_HUD::UpdateMatchTimeLeft(const float TimeLeft)
 {
-	if(IsValid(CharacterOverlay))
+	if (MatchState == MatchState::WaitingToStart)
+	{
+		if(IsValid(AnnouncementOverlay))
+		{
+			AnnouncementOverlay->UpdateMatchTimer(TimeLeft);
+		}
+	}
+	else if (MatchState == MatchState::InProgress)
+	{
+		if(IsValid(CharacterOverlay))
+		{
+			CharacterOverlay->UpdateMatchTimer(TimeLeft);
+		}
+	}
+	else if(MatchState == MatchState::Cooldown)
+	{
+		if(IsValid(ResultsOverlay))
+		{
+			ResultsOverlay->UpdateMatchTimer(TimeLeft);
+		}
+	}
+}
```

Comment:
I can't :( I looked for that because I got an error when I did that
https://forums.unrealengine.com/t/is-it-possible-to-use-fname-for-a-switch-in-c/298295